src/replication.c

```diff
@@ -3918,6 +3963,10 @@ void replicationAbortSyncTransfer(void) {
     cleanupTransferResources();
 }
 
+void waitForDiskSaveBioThreadComplete(void) {
+    while (bioPendingJobsOfType(BIO_SAVE_TO_DISK));
```

Comment:
This is a blocking operation and should be avoided in the main thread.  Although this is already being done in the main thread I think, via: `bioDrainWorker`

Reply:
We have to do this to prevent a race condition where a new sync starts before the previous Bio thread finishes.
Busy-waiting is also how it's done for io-threads, see `waitForClient()`

Reply:
Good point - since we switched to blocking mode on the bio thread we can have a deadlock here. I'll remove this, and then the timeouts of `read()` will guarantee that we eventually reach `shouldAbortSave()` which will free the main thread.

Reply:
Blocking the main thread forever is bad practice, even if the child thread is logically guaranteed to finish. 
Maybe timing it out and crashing is the right approach.